torch/utils/_pytree/api/python.py

```diff
@@ -465,12 +491,23 @@ def _deque_unflatten(values: Iterable[Any], context: Context) -> Deque[Any]:
     _deque_unflatten,
     serialized_type_name="collections.deque",
 )
+_private_register_pytree_node(
+    structseq,
+    _structseq_flatten,
+    _structseq_unflatten,
+    serialized_type_name="structseq",
+    to_dumpable_context=_structseq_serialize,
+    from_dumpable_context=_structseq_deserialize,
+)
 
 
 def _get_node_type(tree: Any) -> Any:
-    node_type = type(tree)
-    if node_type not in SUPPORTED_NODES and is_namedtuple_class(node_type):
-        return namedtuple
+    node_type = tree.__class__  # Dynamo complains about using `type(tree)`
```

Comment:
I'm a bit confused about this. The previous commit uses `node_type = type(tree)`. It works fine and passes all CI tests. But the Dynamo asks to change to `node_type = tree.__class__` in the CI test in this PR.

Reply:
Now the new CI test errors ask to change back to `node_type = type(tree)`.

Use `node_type = type(tree)`, will get error:

<details>
<summary>Error outputs</summary>

```console
$ python test/dynamo/test_higher_order_ops.py -k test_access_module_attr
[2023-11-08 23:32:18,089] [0/0] torch._dynamo.variables.higher_order_ops: [WARNING] speculate_subgraph: while introspecting wrap, we were unable to trace function `NestedUserFunctionVariable` into a single graph. This means that Dynamo was unable to prove safety for this API and will fall back to eager-mode PyTorch, which could lead to a slowdown.
[2023-11-08 23:32:18,089] [0/0] torch._dynamo.variables.higher_order_ops: [ERROR] Can't access members of type(obj) for a generated custom object. Please use __class__ instead
[2023-11-08 23:32:18,089] [0/0] torch._dynamo.variables.higher_order_ops: [ERROR] For more information about this error, see: https://pytorch.org/docs/main/generated/exportdb/index.html#type-reflection-method
Einline_call [("Can't access members of type(obj) for a generated custom object. Please use __class__ instead\nFor more information about this error, see: https://pytorch.org/docs/main/generated/exportdb/index.html#type-reflection-method", 1)]
unimplemented [("speculate_subgraph: while introspecting wrap, we were unable to trace function `NestedUserFunctionVariable` into a single graph. This means that Dynamo was unable to prove safety for this API and will fall back to eager-mode PyTorch, which could lead to a slowdown. Scroll up for the stack trace of the initial exception. The reason was: Can't access members of type(obj) for a generated custom object. Please use __class__ instead\nFor more information about this error, see: https://pytorch.org/docs/main/generated/exportdb/index.html#type-reflection-method", 1)]

======================================================================
ERROR: test_access_module_attr (__main__.HigherOrderOpTests.test_access_module_attr)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/higher_order_ops.py", line 229, in speculate_subgraph
    tree_flatten_output = tree_flatten.call_function(tx, [output], {})
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1298, in LOAD_ATTR
    result = BuiltinVariable(getattr).call_function(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/builtin.py", line 614, in call_function
    result = handler(tx, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/builtin.py", line 1151, in call_getattr
    return obj.var_getattr(tx, name).clone(source=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/base.py", line 262, in var_getattr
    value = self.const_getattr(tx, name)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/constant.py", line 122, in const_getattr
    raise UserError(
torch._dynamo.exc.UserError: Can't access members of type(obj) for a generated custom object. Please use __class__ instead
For more information about this error, see: https://pytorch.org/docs/main/generated/exportdb/index.html#type-reflection-method

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2536, in wrapper
    method(*args, **kwargs)
  File "/Users/PanXuehai/Projects/pytorch/test/dynamo/test_higher_order_ops.py", line 1830, in test_access_module_attr
    result = f(x)
             ^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py", line 409, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py", line 570, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 377, in _convert_frame_assert
    return _compile(
           ^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 594, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/utils.py", line 222, in time_wrapper
    r = func(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 511, in compile_inner
    out_code = transform_code_object(code, transform)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/bytecode_transformation.py", line 1033, in transform_code_object
    transformations(instructions, code_options)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 150, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 476, in transform
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2120, in run
    super().run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/higher_order_ops.py", line 1227, in call_function
    p_args, p_kwargs, example_value, treespec = self.create_wrapped_node(
                                                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/higher_order_ops.py", line 1186, in create_wrapped_node
    ) = speculate_subgraph(
        ^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/higher_order_ops.py", line 288, in speculate_subgraph
    raise Unsupported(
torch._dynamo.exc.Unsupported: speculate_subgraph: while introspecting wrap, we were unable to trace function `NestedUserFunctionVariable` into a single graph. This means that Dynamo was unable to prove safety for this API and will fall back to eager-mode PyTorch, which could lead to a slowdown. Scroll up for the stack trace of the initial exception. The reason was: Can't access members of type(obj) for a generated custom object. Please use __class__ instead
For more information about this error, see: https://pytorch.org/docs/main/generated/exportdb/index.html#type-reflection-method

from user code:
   File "/Users/PanXuehai/Projects/pytorch/test/dynamo/test_higher_order_ops.py", line 1828, in f
    return wrap(lambda y: y - mod.bias, y)

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True


To execute this test, run the following from the base repo dir:
     python test/dynamo/test_higher_order_ops.py -k test_access_module_attr

This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0

----------------------------------------------------------------------
Ran 1 test in 0.089s

FAILED (errors=1)
```

</details>

------

Use `node_type = tree.__class__`, will get error:

<details>
<summary>Error outputs</summary>

```console
$ python test/dynamo/test_misc.py -k test_tracing_tree_map_only
Einline_call [('call_method ConstDictVariable() __contains__ [GetAttrVariable(ConstDictVariable(), __class__)] {}', 1)]

======================================================================
ERROR: test_tracing_tree_map_only (__main__.MiscTests.test_tracing_tree_map_only)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/PanXuehai/Projects/pytorch/torch/testing/_internal/common_utils.py", line 2536, in wrapper
    method(*args, **kwargs)
  File "/Users/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 7174, in test_tracing_tree_map_only
    comp_out = torch._dynamo.optimize(counter, nopython=True)(fn)(xsb)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py", line 409, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/eval_frame.py", line 570, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 377, in _convert_frame_assert
    return _compile(
           ^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 594, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/utils.py", line 222, in time_wrapper
    r = func(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 511, in compile_inner
    out_code = transform_code_object(code, transform)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/bytecode_transformation.py", line 1033, in transform_code_object
    transformations(instructions, code_options)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 150, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/convert_frame.py", line 476, in transform
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2120, in run
    super().run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 469, in wrapper
    return inner_fn(self, inst)
           ^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1799, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 650, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 248, in call_function
    return super().call_function(tx, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/functions.py", line 81, in call_function
    return tx.inline_user_function_return(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 685, in inline_user_function_return
    return InliningInstructionTranslator.inline_call(self, fn, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2252, in inline_call
    return cls.inline_call_(parent, func, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 2384, in inline_call_
    tracer.run()
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 815, in run
    and self.step()
        ^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 778, in step
    getattr(self, inst.opname)(inst)
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/symbolic_convert.py", line 1639, in CONTAINS_OP
    self.push(right.call_method(self, "__contains__", [left], {}))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/dicts.py", line 183, in call_method
    return super().call_method(tx, name, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/variables/base.py", line 321, in call_method
    raise unimplemented(f"call_method {self} {name} {args} {kwargs}")
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/PanXuehai/Projects/pytorch/torch/_dynamo/exc.py", line 192, in unimplemented
    raise Unsupported(msg)
torch._dynamo.exc.Unsupported: call_method ConstDictVariable() __contains__ [GetAttrVariable(ConstDictVariable(), __class__)] {}

from user code:
   File "/Users/PanXuehai/Projects/pytorch/test/dynamo/test_misc.py", line 7166, in fn
    y = pytree.tree_map_only(torch.Tensor, mapper, xs)
  File "/Users/PanXuehai/Projects/pytorch/torch/utils/_pytree/api/python.py", line 761, in tree_map_only
    return tree_map(map_only(__type_or_types)(func), tree)
  File "/Users/PanXuehai/Projects/pytorch/torch/utils/_pytree/api/python.py", line 652, in tree_map
    flat_args, spec = tree_flatten(tree)
  File "/Users/PanXuehai/Projects/pytorch/torch/utils/_pytree/api/python.py", line 587, in tree_flatten
    spec = _tree_flatten_helper(tree, leaves)
  File "/Users/PanXuehai/Projects/pytorch/torch/utils/_pytree/api/python.py", line 568, in _tree_flatten_helper
    if _is_leaf(tree):
  File "/Users/PanXuehai/Projects/pytorch/torch/utils/_pytree/api/python.py", line 516, in _is_leaf
    return _get_node_type(tree) not in SUPPORTED_NODES
  File "/Users/PanXuehai/Projects/pytorch/torch/utils/_pytree/api/python.py", line 506, in _get_node_type
    if node_type not in SUPPORTED_NODES:

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True


To execute this test, run the following from the base repo dir:
     python test/dynamo/test_misc.py -k test_tracing_tree_map_only

This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0

----------------------------------------------------------------------
Ran 1 test in 0.447s

FAILED (errors=1)
```

</details>

Reply:
I found that this error was caused by Dynamo not yet implementing `cls.__base__`.